docker/mongodb-kubernetes-tests/tests/search/search_community_tls.py

```diff
@@ -0,0 +1,324 @@
+            x509.NameAttribute(NameOID.COMMON_NAME, f"{resource_name}.{namespace}.svc.cluster.local"),
+        ]
+    )
+
+    # Create SAN list based on resource type
+    if resource_type == "mdbc":
+        # MongoDBCommunity StatefulSet pods and services
+        san_list = [
+            x509.DNSName(f"{resource_name}-0.{resource_name}-svc.{namespace}.svc.cluster.local"),
+            x509.DNSName(f"{resource_name}-1.{resource_name}-svc.{namespace}.svc.cluster.local"),
+            x509.DNSName(f"{resource_name}-2.{resource_name}-svc.{namespace}.svc.cluster.local"),
+            x509.DNSName(f"{resource_name}-svc.{namespace}.svc.cluster.local"),
+        ]
+    elif resource_type == "mdbs":
+        # MongoDBSearch service names
+        san_list = [
+            x509.DNSName(f"{resource_name}-search-svc.{namespace}.svc.cluster.local"),
+        ]
+
+    server_cert = (
+        x509.CertificateBuilder()
+        .subject_name(server_subject)
+        .issuer_name(ca_cert.subject)
+        .public_key(server_private_key.public_key())
+        .serial_number(x509.random_serial_number())
+        .not_valid_before(datetime.datetime.utcnow())
+        .not_valid_after(datetime.datetime.utcnow() + datetime.timedelta(days=365))
+        .add_extension(
+            x509.SubjectAlternativeName(san_list),
+            critical=False,
+        )
+        .add_extension(
+            x509.BasicConstraints(ca=False, path_length=None),
+            critical=True,
+        )
+        .add_extension(
+            x509.KeyUsage(
+                digital_signature=True,
+                key_encipherment=True,
+                key_agreement=False,
+                key_cert_sign=False,
+                crl_sign=False,
+                data_encipherment=False,
+                content_commitment=False,
+                encipher_only=False,
+                decipher_only=False,
+            ),
+            critical=True,
+        )
+        .add_extension(
+            x509.ExtendedKeyUsage(
+                [
+                    x509.oid.ExtendedKeyUsageOID.SERVER_AUTH,
+                    x509.oid.ExtendedKeyUsageOID.CLIENT_AUTH,
+                ]
+            ),
+            critical=True,
+        )
+        .sign(ca_private_key, hashes.SHA256())
+    )
+
+    return {
+        "cert": server_cert.public_bytes(serialization.Encoding.PEM).decode("utf-8"),
+        "key": server_private_key.private_bytes(
+            encoding=serialization.Encoding.PEM,
+            format=serialization.PrivateFormat.PKCS8,
+            encryption_algorithm=serialization.NoEncryption(),
+        ).decode("utf-8"),
+    }
+
+
+@fixture(scope="module")
+def mdbc(namespace: str) -> MongoDBCommunity:
+    resource = MongoDBCommunity.from_yaml(
+        yaml_fixture("community-replicaset-sample-mflix.yaml"),
+        name=MDBC_RESOURCE_NAME,
+        namespace=namespace,
+    )
+
+    # Add TLS configuration
```

Comment:
please add:
if tryLoad(resource): return resource 

and change CR fixture scope to function. It's a pattern allowing loading (and thus refreshing) the CR with each test's invocation if it's already created.